server.go

```diff
@@ -1187,6 +1191,26 @@ func (ctx *RequestCtx) TimeoutErrorWithResponse(resp *Response) {
 	ctx.timeoutResponse = respCopy
 }
 
+// NextProto adds nph to be processed when key is negotiated when TLS conection is stablished.
+func (s *Server) NextProto(key string, nph ServeHandler) {
+	if s.nextProtos == nil {
+		s.nextProtos = make(map[string]ServeHandler)
+	}
+	if s.tlsConfig == nil {
+		s.tlsConfig = &tls.Config{}
+	}
+	s.tlsConfig.NextProtos = append(s.tlsConfig.NextProtos, key)
+	s.nextProtos[key] = nph
```

Comment:
Maps and slices are not a threadsafe types. We should either document that this func is not concurrency-safe, or add a mutex here

Reply:
@erikdubbelboer yes. It must be called before *Serve functions.